pkg/combination/combination.go

```diff
@@ -58,28 +75,76 @@ func WithSolveAhead() StreamOption {
 }
 
 // Next gets the next combination from the stream.
-// TODO: Currently this does not solve each combination iteratively and will
-//       need to do so in the future to ensure an optimal use of memory. This
-//       is currently more of a stub to allow consumer packages to maintain its
-// 		 interface.
+// By using this, the stream will solve each combination
+// iteratively.
```

Comment:
@tylerslaton Should this be opt-in behavior? Should we expose some configuration knobs such that a user can specify how they'd like to have stream combinations be generated?

Reply:
Yeah, good call. I think it is good to define this with the `solveAhead` property.

Reply:
@jbuchananr to do this you could break out the `Next()` into two smaller private functions.

`nextIterative()` What the old function was
`nextAll()` - What your current function is

Then you would just call one or the other depending on the value of `cs.solveAhead`

Reply:
I think we might also be reaching complexity creep with this method now too. I see a lot of mutation around the `cs.solved` field, which is fine for now, but is likely a candidate for refactoring down the line.